libpod/in_memory_state.go

```diff
@@ -244,7 +339,13 @@ func (s *InMemoryState) ContainerInUse(ctr *Container) ([]string, error) {
 func (s *InMemoryState) AllContainers() ([]*Container, error) {
 	ctrs := make([]*Container, 0, len(s.containers))
 	for _, ctr := range s.containers {
-		ctrs = append(ctrs, ctr)
+		if s.namespace != "" {
+			if ctr.config.Namespace == s.namespace {
+				ctrs = append(ctrs, ctr)
+			}
+		} else {
+			ctrs = append(ctrs, ctr)
```

Comment:
(`if s.namespace == "" || ctr.config.Namespace == s.namespace` would allow having a single `if` instead of `if`/`if`//`else`. Same in `AllPods`.)